nixos/doc/manual/development/running-nixos-tests-interactively.section.md

```diff
@@ -90,6 +107,35 @@ $ ./result/bin/nixos-test-driver --keep-vm-state
 The machine state is stored in the `$TMPDIR/vm-state-machinename`
 directory.
 
+## Rebuilding the test / redeploying the VMs {#sec-nixos-test-rebuild}
+
+When you change the VM configurations, you can update the running interactive
+driver and redeploy to those machines without restarting. The `rebuild()`
+command at the python REPL will update the running driver based on the
+executable produced by the command provided.
+
+```py
+>>> rebuild("nix-build . -A nixosTests.login.driverInteractive")
+```
+
+By default the new `nixos-test-driver` executable is assumed to be located at
+the same place the running driver was launched from. This works when the
+`result/` symlink is updated to point at the new derivation output. If needed, the path
+can be overridden with the `exe` parameter to `rebuild()`.
+
+You can also specify the rebuild command at the command line when the driver is
+first launched. Then `rebuild()` can be called without arguments.
+
+```ShellSession
+$ build_cmd="nix-build . -A nixosTests.login.driverInteractive"
+$ eval $build_cmd
+$ ./result/bin/nixos-test-driver --rebuild-cmd $build_cmd 2>machine_output
```

Comment:
Good call on the quotes.

What stderr output are you worried about? Python errors, for instance, are output to stdout.

I added it to the docs because in my experience the interactive driver is extremely irritating to use with the VM output clogging up the interactive terminal, and it took me a while to realise I could redirect it.

Reply:
Oh indeed calls to `logger.info()` and the like get sent to stderr. I'll just remove this, since it's not actually related to the PR.